homeassistant/components/device_tracker/icloud.py

```diff
@@ -19,49 +20,116 @@
+        accountname = call.data.get('accountname')
+        entity_id = 'icloud.' + accountname
+        devicename = call.data.get('devicename')
+        if entity_id in ICLOUDTRACKERS:
+            ICLOUDTRACKERS[entity_id].lost_iphone(devicename)
+
+    hass.services.register('device_tracker', 'lost_iphone',
+                           lost_iphone)
+
+    def update_icloud(call):
+        """ Calls the update function of an icloud account """
+        accountname = call.data.get('accountname')
+        entity_id = 'icloud.' + accountname
+        if entity_id in ICLOUDTRACKERS:
+            ICLOUDTRACKERS[entity_id].update_icloud(see)
+
+    hass.services.register('device_tracker',
+                           'update_icloud', update_icloud)
+    iclouddevice.update_icloud(see)
 
-    if username is None or password is None:
-        _LOGGER.error('Must specify a username and password')
-        return False
+    track_utc_time_change(
+        hass, iclouddevice.update_icloud(see),
+        minute=range(0, 60, config.get(CONF_INTERVAL, DEFAULT_INTERVAL)),
+        second=0
+    )
+
+    return True
 
-    try:
-        _LOGGER.info('Logging into iCloud Account')
-        # Attempt the login to iCloud
-        api = PyiCloudService(username,
-                              password,
-                              verify=True)
-    except PyiCloudFailedLoginException as error:
-        _LOGGER.exception('Error logging into iCloud Service: %s', error)
-        return False
 
-    def keep_alive(now):
-        """ Keeps authenticating iCloud connection. """
-        api.authenticate()
-        _LOGGER.info("Authenticate against iCloud")
+class Icloud(Entity):  # pylint: disable=too-many-instance-attributes
+    """ Represents a Proximity in Home Assistant. """
+    def __init__(self, hass, username, password, name):
+        # pylint: disable=too-many-arguments
+        from pyicloud import PyiCloudService
+        from pyicloud.exceptions import PyiCloudFailedLoginException
 
-    track_utc_time_change(hass, keep_alive, second=0)
+        self.hass = hass
+        self.username = username
+        self.password = password
+        self.accountname = name
+        self.api = None
 
-    def update_icloud(now):
+        if self.username is None or self.password is None:
+            _LOGGER.error('Must specify a username and password')
+        else:
+            try:
+                _LOGGER.info('Logging into iCloud Account')
+                # Attempt the login to iCloud
+                self.api = PyiCloudService(self.username,
+                                           self.password,
+                                           verify=True)
+            except PyiCloudFailedLoginException as error:
+                _LOGGER.exception('Error logging into iCloud Service: %s',
+                                  error)
+
+    @property
+    def state(self):
+        return self.api is not None
+
+    @property
+    def state_attributes(self):
```

Comment:
See, if this was being used, this would have been a **horrible** idea. Publishing authentication info :warning: . Just remove it.